src/main/webapp/nav.html

```diff
@@ -4,13 +4,13 @@
       <img src="images/logo.jpg" class="rounded-circle" alt="logo" height="80" width="100">
     </li>
     <li class="nav-item">
-      <a class="nav-link nav-text-style" href="profile.html">Profile</a>
+      <a class="nav-link nav-text-style" href="#" onclick="setProfileLink()" id="profileLink">Profile</a>
```

Comment:
Instead of changing the `href` in `#onClick`, which will break some behavior (e.g. Right-click -> Open in new tab or Copy Link Address), it'd be better to do it when the page first loads.

Reply:
Followed-up offline, tried out MutationObserver but couldn't get it to work. Fine with punting on this to avoid blocking you!